modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -129,6 +129,101 @@ G_TYPED_KERNEL(GBackgroundSubtractor, <GMat(GMat, BackgroundSubtractorParams)>,
     }
 };
 
+/** @brief Structure for the Kalman filter's initialization parameters.*/
+
+struct KalmanParams
+{
+    //! Type of the created matrices that should be CV_32F or CV_64F.
+    int type;
+    //! Dimensionality of the control vector.
+    int ctrlDim;
+    //! Dimensionality of the state.
+    int dpDim;
+    //! Dimensionality of the measurement.
+    int mpDim;
+
+    // initial state
+
+    //! predicted state(x'(k)): x(k)=A*x(k-1)+B*u(k)
+    Mat statePre;
+    //! priori error estimate covariance matrix (P'(k)): P'(k)=A*P(k-1)*At + Q)*/
+    Mat errorCovPre;
+
+    // dynamic system description
+
+    //! state transition matrix (A)
+    Mat transitionMatrix;
+    //! measurement matrix (H)
+    Mat measurementMatrix;
+    //! process noise covariance matrix (Q)
+    Mat processNoiseCov;
+    //! measurement noise covariance matrix (R)
+    Mat measurementNoiseCov;
+    //! control matrix (B) (Optional: not used if there's no control)
+    Mat controlMatrix;
+
+    /** Full constructor
+    @param dp Dimensionality of the state.
+    @param mp Dimensionality of the measurement.
+    @param cp Dimensionality of the control vector. If it equals 0, dynamic system don't have external impact.
+    So controlMatrix should be empty.
+    @param depth Type of the created matrices that should be CV_32F or CV_64F.
+    */
+    KalmanParams(int dp, int mp, int cp = 0, int depth = CV_32F)
+    {
+        GAPI_Assert(dp > 0 && mp > 0);
+        GAPI_Assert(depth == CV_32F || depth == CV_64F);
+        ctrlDim = std::max(cp, 0);
+        type = depth;
+        dpDim = dp;
+        mpDim = mp;
+
+        statePre = Mat::zeros(dpDim, 1, type);
+        transitionMatrix = Mat::eye(dpDim, dpDim, type);
+
+        processNoiseCov = Mat::eye(dpDim, dpDim, type);
+        measurementMatrix = Mat::zeros(mpDim, dpDim, type);
+        measurementNoiseCov = Mat::eye(mpDim, mpDim, type);
+
+        errorCovPre = Mat::zeros(dpDim, dpDim, type);
+
+        if (ctrlDim > 0)
+            controlMatrix = Mat::zeros(dpDim, ctrlDim, type);
+        else
+            controlMatrix.release();
+    }
+};
+
+G_TYPED_KERNEL(GKalmanFilter, <GMat(GMat, GOpaque<bool>, GMat, KalmanParams)>, "org.opencv.video.KalmanFilter")
+{
+    static GMatDesc outMeta(const GMatDesc& measurement, const GOpaqueDesc&, const GMatDesc&, const KalmanParams& kfParams)
+    {
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        GAPI_Assert(kfParams.dpDim > 0 && kfParams.mpDim > 0 && kfParams.ctrlDim >= 0);
+
+        GAPI_Assert(kfParams.statePre.depth() == kfParams.type &&
```

Comment:
In OpenCV, `depth` and `type` are different things. I assume you should compare `depth` against `depth` or `type` against `type`. `type` includes channel information, `depth` doesn't.

Reply:
Ok. Changed.